webbrowser/open_browser_unix.go

```diff
@@ -9,27 +9,40 @@ import (
 	"os/exec"
 )
 
-// OpenInBrowser checks for the presence of a display server
-// and environment variables indicating a gui is present. If found
-// then xdg-open is called on a url to open said url in the default
-// gui web browser for the system
+// Open opens `url` in default system browser. It tries to do so in two
+// ways (xdg-open and $BROWSER). It only works if there is a display
+// server working.
+//
+// bouncepaw: I tried to support TTYs as well. The idea was to open
+// a browser in foreground and return back to amfora after the browser
+// is closed. While all browsers I tested opened correctly (w3m, lynx),
+// I couldn't make it restore amfora correctly. The screen always ended
+// up distorted. None of my stunts with altscreen buffers helped.
 func Open(url string) (string, error) {
-	disp := os.Getenv("DISPLAY")
-	wayland := os.Getenv("WAYLAND_DISPLAY")
-	_, err := exec.LookPath("Xorg")
-	if disp == "" && wayland == "" && err != nil {
-		return "", fmt.Errorf("no gui is available")
+	var (
+		// In prev versions there was also Xorg executable checked for.
+		// I don't see any reason to check for it.
+		xorgDisplay                     = os.Getenv("DISPLAY")
+		waylandDisplay                  = os.Getenv("WAYLAND_DISPLAY")
+		xdgOpenPath, xdgOpenNotFoundErr = exec.LookPath("xdg-open")
+		envBrowser                      = os.Getenv("BROWSER")
+	)
+	switch {
+	case xorgDisplay == "" && waylandDisplay == "":
+		return "", fmt.Errorf("no display server was found")
+	case xdgOpenNotFoundErr == nil: // Prefer xdg-open over $BROWSER
+		// Use start rather than run or output in order
+		// to make browser running in background.
+		if err := exec.Command(xdgOpenPath, url).Start(); err != nil {
+			return "", err
+		}
+		return "Opened in system default web browser", nil
+	case envBrowser != "":
+		if err := exec.Command(xdgOpenPath, url).Start(); err != nil {
```

Comment:
```suggestion
		if err := exec.Command(envBrowser, url).Start(); err != nil {
```

Please also see my [other comment](https://github.com/makeworld-the-better-one/amfora/pull/93#discussion_r498361088).

Reply:
Well, I'm not very attentive today, as you've noticed. Thanks for thorough review